riscv_isac/InstructionObject.py

```diff
@@ -466,11 +476,28 @@ def trap_registers_update(self, instr_vars, trap_dict):
         : param instr_vars: Dictionary holding the values of current instruction state
         : param trap_dict : Values for the trap registers for current instruction 
         '''
+
         instr_vars['mode_change']   = trap_dict['mode_change']
-        instr_vars['exc_num']       = trap_dict['exc_num']
         instr_vars['call_type']     = trap_dict['call_type']
-        instr_vars['tval']          = trap_dict['tval']
-        
+
+        if trap_dict["mode_change"] is not None:
+            #update the registers depending upon the mode change
+            if trap_dict["mode_change"].split()[2] == "M":
+                instr_vars['mcause']      = trap_dict['exc_num']
+                instr_vars['mtval']       = trap_dict['tval']
+                instr_vars['scause']      = '0'
```

Comment:
Stupid question: does this clear scause when the mode change is to Mmode? Because it should just leave it unchanged...

Reply:
It will leave the scause unchanged, @MuhammadHammad001 can give more details but looking at the code, I think scause will remain unchanged

Reply:
Well, I might believe the actual scause doesn't change. The question is whether ISAC's tracking of the scause value makes it think that it is now zero, so it messes up coverpoint matching (I think)

Reply:
Yes, the values for mcause, scause, mtval and stval will be maintained until they are updated again by the exception/call.
P.S: I added the comment few days ago, but github made it into `pending`  which made it not visible to others.